subsys/logging/CMakeLists.txt

```diff
@@ -14,6 +14,15 @@ if(NOT CONFIG_LOG_MINIMAL)
     log_backend_uart.c
   )
 
+  zephyr_sources_ifdef(
+    CONFIG_LOG_BACKEND_FS
+    log_backend_fs.c
+  )
+
+  zephyr_include_directories(
+    ${ZEPHYR_BASE}/../modules/fs/littlefs
```

Comment:
is it needed here. doesn't it come with littlefs being enabled?

Reply:
There is an error without above code inserted and `CONFIG_FILE_SYSTEM_LITTLEFS=y` set: `lfs.h No such file or directory`.

Reply:
still, if there is something wrong it should not be solved here.
There is a `CMakeLists.txt` in `modules/fs/littlefs`. Maybe it is something wrong there?
```
target_include_directories(LITTLEFS INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
```
@pabigot as you were the one adding it, do you have any ideas?

Reply:
The file systems are packaged into libraries that provide all the flags and options required to make use of them.

Following the shell subsystem I think the correct fix is to add something like this:

  target_link_libraries_ifdef(CONFIG_FAT_FILESYSTEM_ELM   LOG INTERFACE ELMFAT)
  target_link_libraries_ifdef(CONFIG_FILE_SYSTEM_LITTLEFS LOG INTERFACE LITTLEFS)

but it appears that requires making the log subsystem an interface library like the shell subsystem is.  There may be a better solution; a CMake expert like @tejlmand can provide more informed help.